packages/lesswrong/server/languageModels/languageModelIntegration.ts

```diff
@@ -0,0 +1,307 @@
+import { Globals } from '../../lib/vulcan-lib/config';
+import { Configuration as OpenAIApiConfiguration, OpenAIApi } from "openai";
+import { Tags } from '../../lib/collections/tags/collection';
+import { dataToMarkdown } from '../editor/conversionUtils';
+import { DatabaseServerSetting } from '../databaseSettings';
+import { encode as gpt3encode, decode as gpt3decode } from 'gpt-3-encoder'
+import drop from 'lodash/drop';
+import take from 'lodash/take';
+
+const openAIApiKey = new DatabaseServerSetting<string|null>('languageModels.openai.apiKey', null);
+const openAIOrganizationId = new DatabaseServerSetting<string|null>('languageModels.openai.organizationId', null);
+
+let openAIApi: OpenAIApi|null = null;
+export async function getOpenAI(): Promise<OpenAIApi|null> {
+  if (!openAIApi){
+    const apiKey = openAIApiKey.get();
+    const organizationId = openAIOrganizationId.get();
+    
+    if (apiKey) {
+      openAIApi = new OpenAIApi(new OpenAIApiConfiguration({
+        apiKey,
+        organization: organizationId ?? undefined,
+      }));
+    }
+  }
+  return openAIApi;
+}
+
```

Comment:
Not the place for it, but this reminds me that we need to think about how to handle initializing singletons on server startup.  (In this case we might not have a better way to do it since it can definitely be missing for the EA forum, of course.)